OBSDOCS-873/TRACING-3799: Documentation for the Tempo monolithic CRD (#75761)
openshift/openshift-docs: modules/distr-tracing-tempo-install-tempostack-web-console.adoc

:_mod-docs-content-type: PROCEDURE
[id="distr-tracing-tempo-install-tempostack-web-console_{context}"]
= Installing a TempoStack instance by using the web console

You can install a TempoStack instance from the *Administrator* view of the web console.

.Prerequisites

* You are logged in to the {product-title} web console as a cluster administrator with the `cluster-admin` role.

* For {product-dedicated}, you must be logged in using an account with the `dedicated-admin` role.

* You have completed setting up the required object storage by a supported provider: link:https://www.redhat.com/en/technologies/cloud-computing/openshift-data-foundation[Red Hat OpenShift Data Foundation], link:https://min.io/[MinIO], link:https://aws.amazon.com/s3/[Amazon S3], link:https://azure.microsoft.com/en-us/products/storage/blobs/[Azure Blob Storage], link:https://cloud.google.com/storage/[Google Cloud Storage]. For more information, see "Object storage setup".
+
[WARNING]
====
Object storage is required and not included with the {TempoShortName}. You must choose and set up object storage by a supported provider before installing the {TempoShortName}.
====

.Procedure

> . Create a project of your choice for the TempoStack instance that you will create in a subsequent step: go to *Home* -> *Projects* -> *Create Project*.

Reviewer: ```suggestion
. Create a project of your choice for the TempoStack instance that you will create in a subsequent step by clicking *Home* -> *Projects* -> *Create Project*.
```

Author: TY. I'll apply the same solution here as in the other comment.

Author: This is how I rephrased it:

`. Go to *Home* -> *Projects* -> *Create Project* to create a project of your choice for the TempoStack instance that you will create in a subsequent step.`

> . In the project that you created for the TempoStack instance, create a secret for your object storage bucket: go to *Workloads* -> *Secrets* -> *Create* -> *From YAML*. For more information, see "Object storage setup".

Author: TY. I'll apply the same solution here as in the other comment.

> . In the project that you created for the TempoStack instance, create a secret for your object storage bucket: go to *Workloads* -> *Secrets* -> *Create* -> *From YAML*. For more information, see "Object storage setup".

Author: This is how I rewrote it:

```
. Go to *Workloads* -> *Secrets* -> *Create* -> *From YAML* to create
a secret for your object storage bucket in the project that you created for
the TempoStack instance. For more information, see "Object storage setup".
```

+
--
include::snippets/distr-tracing-tempo-secret-example.adoc[]
--

. Create a TempoStack instance.
+
[NOTE]
====
You can create multiple TempoStack instances in separate projects on the same cluster.
====

.. Go to *Operators* -> *Installed Operators*.

.. Select *TempoStack* -> *Create TempoStack* -> *YAML view*.

.. In the *YAML view*, customize the `TempoStack` custom resource (CR):
+
[source,yaml]
----
apiVersion: tempo.grafana.com/v1alpha1
kind: TempoStack
metadata:
  name: sample
  namespace: <project_of_tempostack_instance>
spec:
  storageSize: 1Gi
  storage:
    secret: # <1>

>       name: <secret-name> # <2>
>       type: <secret-provider> # <3>

Reviewer 2: ```suggestion
      name: <secret_name> # <2>
      type: <secret_provider> # <3>
```

Author: Thank you. This one is embarrassing.
